src/main/java/com/sprint/mission/discodeit/service/basic/BasicMessageService.java

```diff
@@ -0,0 +1,122 @@
+package com.sprint.mission.discodeit.service.basic;
+
+import com.sprint.mission.discodeit.dto.binary_content.BinaryContentCreateRequest;
+import com.sprint.mission.discodeit.dto.message.MessageCreateRequest;
+import com.sprint.mission.discodeit.dto.message.MessageDTO;
+import com.sprint.mission.discodeit.dto.message.MessageUpdateRequest;
+import com.sprint.mission.discodeit.entity.BinaryContent;
+import com.sprint.mission.discodeit.entity.Channel;
+import com.sprint.mission.discodeit.entity.ChannelType;
+import com.sprint.mission.discodeit.entity.Message;
+import com.sprint.mission.discodeit.entity.User;
+import com.sprint.mission.discodeit.repository.BinaryContentRepository;
+import com.sprint.mission.discodeit.repository.ChannelRepository;
+import com.sprint.mission.discodeit.repository.MessageRepository;
+import com.sprint.mission.discodeit.repository.UserRepository;
+import com.sprint.mission.discodeit.service.BinaryContentService;
+import com.sprint.mission.discodeit.service.ChannelService;
+import com.sprint.mission.discodeit.service.MessageService;
+import jakarta.transaction.Transactional;
+import java.io.IOException;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+import org.springframework.web.multipart.MultipartFile;
+
+@RequiredArgsConstructor
+@Service
+public class BasicMessageService implements MessageService {
+
+  private final MessageRepository messageRepository;
+  private final ChannelRepository channelRepository;
+  private final UserRepository userRepository;
+  private final BinaryContentRepository binaryContentRepository;
+  private final BinaryContentService binaryContentService;
+  private final ChannelService channelService;
+
+  @Transactional
+  @Override
+  public Message create(MessageCreateRequest messageCreateRequest,
+      List<MultipartFile> attachments) {
+    Channel channel = channelRepository.findById(messageCreateRequest.channelId())
+        .orElseThrow(() -> new NoSuchElementException("채널이 존재하지 않습니다."));
+
+    User writer = userRepository.findById(messageCreateRequest.writerId())
+        .orElseThrow(() -> new NoSuchElementException("유저가 존재하지 않습니다."));
+
+    boolean isMember = channelService.getParticipantIds(channel.getId()).stream()
+        .anyMatch(uuid -> uuid.equals(messageCreateRequest.writerId()));
+
+    if (channel.getType() == ChannelType.PRIVATE && !isMember) {
+      throw new IllegalArgumentException("PRIVATE 채널의 멤버가 아닙니다.");
```

Comment:
isMember인지 조회하는 로직을 if문 안으로 넣으면 불필요한 쿼리를 줄일 수 있을 것 같아요.

```suggestion
    if (channel.getType() == ChannelType.PRIVATE) {
         boolean isMember = channelService.getParticipantIds(channel.getId()).stream()
             .anyMatch(uuid -> uuid.equals(messageCreateRequest.writerId()));
         throw new IllegalArgumentException("PRIVATE 채널의 멤버가 아닙니다.");
```